lib/smack/SmackRep.cpp

```diff
@@ -514,14 +516,27 @@ const Stmt* SmackRep::returnValueAnnotation(const CallInst& CI) {
 //
 // }
 
+bool SmackRep::isUnsafeFloatAccess(const Type* elemTy, const Type* resultTy) {
+  if (elemTy->isFloatingPointTy()) {
+    if (!resultTy || (resultTy->isIntegerTy() && resultTy->getIntegerBitWidth() == 8UL)) {
+      if (!SmackOptions::BitPrecise)
+        return true;
+    } else
+      assert(resultTy->isFloatingPointTy() && "Unsupported map result type.");
```

Comment:
Hmmm, it is weird to have just an assert statement in this else branch. Could this somehow be rewritten?